src/SqlBinding/SqlBindingUtilities.cs

```diff
@@ -173,21 +175,31 @@ public static void GetTableAndSchema(string fullName, out string quotedSchema, o
             // ensure names are properly escaped
             string escapedFullName = fullName.Replace("'", "''");
 
-            // defaults
-            quotedTableName = $"'{escapedFullName}'";
-            quotedSchema = "SCHEMA_NAME()"; // default to user schema
+            var parser = new TSql150Parser(false);
+            var stringReader = new StringReader(escapedFullName);
+            SchemaObjectName tree = parser.ParseSchemaObjectName(stringReader, out IList<ParseError> errors);
 
-            // remove [ ] from name if necessary
-            string cleanName = escapedFullName.Replace("]", string.Empty).Replace("[", string.Empty);
-
-            // if in format schema.table, split into two parts for query
-            string[] pieces = cleanName.Split('.');
-
-            if (pieces.Length == 2)
+            foreach (ParseError err in errors)
             {
-                quotedSchema = $"'{pieces[0]}'";
-                quotedTableName = $"'{pieces[1]}'";
+                Console.WriteLine(err.Message);
```

Comment:
If there's errors I'm not sure we should be continuing

(and even if we were this should be `Console.WriteError`)